src/main/java/seedu/address/logic/commands/FilterCommand.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.util.FilterUtil;
+import seedu.address.model.Model;
+import seedu.address.model.filter.FilterByContainsFilterWordPredicate;
+
+/**
+ * Filters all pets in address book by field provided.
+ * Filter word matching is case-insensitive.
+ * Fields can only be from a defined set.
+ */
+public class FilterCommand extends Command {
+
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters all pets by "
+            + "the specified field followed by filter/ search word (case-insensitive)"
+            + "and displays them as a list with index numbers.\n"
+            + "PARAMETERS: FIELD [FILTER WORD] \n"
+            + "FIELD = { 'byDate/', 'byOwner/', 'byTag/', byApp/ } \n"
```

Comment:
At first I thought of using your suggestion or something similar. However, halfway through the implementation, it felt that there were too many fields to take note off and I thought that having a prefix that is not reused (in the case of owner, tag, app) will be much more straightforward. And the point of making it byDate, byOwner is so that when the user types in `filter byDate/...` it sounds very natural and intuitive compared to `filter by/ date/ ...`which feels more fragmented and prone to some confusion or bugs if not implemented carefully. In terms of scalability, I think that the prefixes may not make much of a difference because we might only be reducing the amount of variables and code in `CliSyntax.java`. We will still need to implement a predicate class for every field we want to filter by and the bulk of the code comes from there. So by packaging these predicate classes under the filter package, it makes it neater and more comprehensive for 'future developers' to scale the feature.